protocol/communities/manager.go

```diff
@@ -1370,65 +1371,144 @@ func (m *Manager) HandleCommunityEventsMessage(signer *ecdsa.PublicKey, message
 
 func (m *Manager) handleAdditionalAdminChanges(community *Community) error {
 
-	saveOrUpdateRequestToJoin := func(signer string, request *protobuf.CommunityRequestToJoin, state RequestToJoinState) error {
-		requestToJoin := &RequestToJoin{
-			PublicKey:        signer,
-			Clock:            request.Clock,
-			ENSName:          request.EnsName,
-			CommunityID:      request.CommunityId,
-			State:            state,
-			RevealedAccounts: request.RevealedAccounts,
-		}
+	if !(community.IsControlNode() || community.HasPermissionToSendCommunityEvents()) {
+		// we're a normal user/member node, so there's nothing for us to do here
+		return nil
+	}
 
-		requestToJoin.CalculateID()
+	for i := range community.config.EventsData.Events {
+		communityEvent := &community.config.EventsData.Events[i]
+		switch communityEvent.Type {
+		case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT:
+			err := m.handleCommunityEventRequestAccepted(community, communityEvent)
+			if err != nil {
+				return err
+			}
 
-		existingRequestToJoin, err := m.persistence.GetRequestToJoin(requestToJoin.ID)
-		if err != nil && err != sql.ErrNoRows {
-			return err
+		case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_REJECT:
+			err := m.handleCommunityEventRequestRejected(community, communityEvent)
+			if err != nil {
+				return err
+			}
+
+		default:
 		}
+	}
+	return nil
+}
 
-		if existingRequestToJoin != nil {
-			// node already knows about this request to join, so let's compare clocks
-			// and update it if necessary
-			if existingRequestToJoin.Clock <= requestToJoin.Clock {
-				pk, err := common.HexToPubkey(existingRequestToJoin.PublicKey)
-				if err != nil {
-					return err
-				}
-				err = m.persistence.SetRequestToJoinState(common.PubkeyToHex(pk), community.ID(), state)
-				if err != nil {
-					return err
-				}
+func (m *Manager) saveOrUpdateRequestToJoin(signer string, communityID types.HexBytes, requestToJoin *RequestToJoin) (bool, error) {
+	updated := false
+
+	existingRequestToJoin, err := m.persistence.GetRequestToJoin(requestToJoin.ID)
+	if err != nil && err != sql.ErrNoRows {
+		return updated, err
+	}
+
+	if existingRequestToJoin != nil {
+		// node already knows about this request to join, so let's compare clocks
+		// and update it if necessary
+		if existingRequestToJoin.Clock <= requestToJoin.Clock {
```

Comment:
```suggestion
		if existingRequestToJoin.Clock < requestToJoin.Clock {
```

I know it was like that before, but I believe same clock should be ignored.

Reply:
As discussed in https://github.com/status-im/status-go/pull/3843/files#r1282298019 we expect it to be the same clock, so we don't want to ignore this